pennylane/templates/subroutines/out_poly.py

```diff
@@ -0,0 +1,357 @@
+            if key
+            not in [
+                "f",
+                "registers_wires",
+            ]
+        )
+        return (
+            self.hyperparameters["f"],
+            self.hyperparameters["registers_wires"],
+        ), metadata
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        hyperparams_dict = dict(metadata)
+        return cls(*data, **hyperparams_dict)
+
+    def map_wires(self, wire_map: dict):
+
+        new_registers_wires = [
+            [wire_map[wire] for wire in input_wire]
+            for input_wire in self.hyperparameters["registers_wires"]
+        ]
+
+        new_work_wires = (
+            [wire_map[wire] for wire in self.hyperparameters["work_wires"]]
+            if self.hyperparameters.get("work_wires")
+            else None
+        )
+
+        return OutPoly(
+            self.hyperparameters["f"],
+            new_registers_wires,
+            mod=self.hyperparameters["mod"],
+            work_wires=new_work_wires,
+        )
+
+    @classmethod
+    def _primitive_bind_call(cls, *args, **kwargs):
+        return cls._primitive.bind(*args, **kwargs)
+
+    def decomposition(self):  # pylint: disable=arguments-differ
+        return self.compute_decomposition(**self.hyperparameters)
+
+    @staticmethod
+    def compute_decomposition(**kwargs):  # pylint: disable=arguments-differ
+
+        f = kwargs["f"]
+        mod = kwargs["mod"]
+
+        registers_wires = kwargs["registers_wires"]
+        work_wires = kwargs["work_wires"]
+
+        list_ops = []
+
+        output_adder_mod = (
+            [work_wires[0]] + registers_wires[-1] if work_wires else registers_wires[-1]
+        )
+        list_ops.append(qml.QFT(wires=output_adder_mod))
+
+        wires_vars = [len(w) for w in registers_wires[:-1]]
+
+        coeffs_list = _get_coefficients_and_controls(f, mod, *wires_vars)
+
+        for item, coeff in coeffs_list.items():
+
+            if not 1 in item:
+                # Add the independent term
+                list_ops.append(qml.PhaseAdder(int(coeff), output_adder_mod))
+                continue
+
+            all_wires_input = sum([*registers_wires[:-1]], start=[])
+
+            controls = [all_wires_input[i] for i, bit in enumerate(item) if bit == 1]
+
+            if work_wires:
+                list_ops.append(
+                    qml.ctrl(
+                        qml.PhaseAdder(
+                            int(coeff) % mod,
+                            output_adder_mod,
+                            work_wire=work_wires[1],
+                            mod=mod,
+                        ),
+                        control=controls,
+                    )
+                )
+            else:
+                list_ops.append(
+                    qml.ctrl(
+                        qml.PhaseAdder(int(coeff) % mod, output_adder_mod),
+                        control=controls,
+                    )
+                )
```

Comment:
I did it this way because I got an error the other way. Digging more into it, I think it may be related to a possible bug in `qml.PhaseAdder`. I'm going to look into it and if so, I will remove the if 👍

Reply:
Bug fixed, the condition has been removed :)